drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -298,7 +284,28 @@ static int32_t optimize_regulator_voltage_scale(uint32_t sysclk_freq)
 	defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI) || \
 	defined(CONFIG_CLOCK_STM32_PLL_SRC_CSI)
 
-static int32_t get_vco_output_range(uint32_t vco_input_range)
+static int get_vco_input_range(uint32_t m_div, uint32_t *range)
+{
+	uint32_t vco_freq;
+
+	vco_freq = PLLSRC_FREQ / m_div;
+
+	if (1000000UL <= vco_freq && vco_freq <= 2000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_1_2;
+	} else if (2000000UL < vco_freq && vco_freq <= 4000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_2_4;
+	} else if (4000000UL < vco_freq && vco_freq <= 8000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_4_8;
+	} else if (8000000UL < vco_freq && vco_freq <= 16000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_8_16;
+	} else {
+		return -ERANGE;
+	}
```

Comment:
```suggestion

#define FREQ_MHz(x) (x*1000000UL)

	if (MHz(1) <= vco_freq && vco_freq <= MHz(2)) {
		*range = LL_RCC_PLLINPUTRANGE_1_2;
	} else if (MHz(2) < vco_freq && vco_freq <= MHz(4)) {
		*range = LL_RCC_PLLINPUTRANGE_2_4;
	} else if (MHz(4) < vco_freq && vco_freq <= MHz(8)) {
		*range = LL_RCC_PLLINPUTRANGE_4_8;
	} else if (MHz(8) < vco_freq && vco_freq <= MHz(16)) {
		*range = LL_RCC_PLLINPUTRANGE_8_16;
	} else {
		return -ERANGE;
	}
```

Reply:
I'd prefer to discuss a generic macro such as this one at Zephyr level, not at module level

Reply:
FYI:
https://github.com/zephyrproject-rtos/zephyr/blob/2d215484cc352152e9686477fead841c13ebb249/include/sys/util.h#L306

Reply:
@martinjaeger great, was not aware of it!

Reply:
fixed